src/Propel/Generator/Model/ForeignKey.php

```diff
@@ -835,6 +835,31 @@ public function isForeignPrimaryKey()
             && !array_diff($foreignPKCols, $foreignCols));
     }
 
+    /**
+     * Returns whether this foreign key is not the primary key of the foreign
+     * table.
+     *
+     * @return bool Returns true if all columns inside this foreign key are not primary keys of the foreign table
+     */
+    public function isForeignNonPrimaryKey()
```

Comment:
Do we want to add `: bool` typehint here?

Reply:
Could you please explain why it is BC? This is similar to above methods `isForeignPrimaryKey`

Reply:
This method does not exist. So anyone that extended this class will not have a problem with the typehint here
Adding it to other existing methods will cause a fatal error on extended classes.